Source/Core/VideoCommon/ShaderGenCommon.h

```diff
@@ -62,6 +59,9 @@ class ShaderGeneratorInterface
 	 */
 	template<class uid_data>
 	uid_data* GetUidData() { return nullptr; }
+
+protected:
+	std::string m_buffer;
```

Comment:
Please reserve a bigger buffer within the constructor. Operator+= isn't the best way to grow this buffer often.